### PR TITLE
keycloak Setup and integration with Score

### PR DESCRIPTION
**Pull request by MitchellShiell**
Required reorg of doc layout plus added images (this is largely why there are 30 files changed). I will post needed song updates in a separate PR. 

This branch is relative to the following ticket: https://app.zenhub.com/workspaces/overture-stack-5d2e058ff67cc800011fee6b/issues/gh/overture-stack/score/395

### CODE REVIEW

File: markdown/documentation/score/installation/configuration/authentication.md [hunk 1]
```diff
@@ -1,69 +1,129 @@
...
+AUTH_SERVER_SCOPE_UPLOAD_SUFFIX=.WRITE
+AUTH_SERVER_SCOPE_DOWNLOAD_SUFFIX=.READ
+AUTH_SERVER_SCOPE_DOWNLOAD_SYSTEM=score.WRITE
+AUTH_SERVER_SCOPE_DOWNLOAD_SUFFIX=.READ
+AUTH_SERVER_SCOPE_UPLOAD_SYSTEM=score.READ
+AUTH_SERVER_SCOPE_UPLOAD_SUFFIX=.WRITE
+SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWKSETURI=http://localhost/realms/myrealm/protocol/openid-connect/certs
+```
+
+<Warning>**Note:** For Keycloak integration, use the Docker container `ghcr.io/overture-stack/score-server:47f006ce`.</Warning>
```
Comment (leoraba) [inline markdown/documentation/score/installation/configuration/authentication.md:92]: this label would soon change to `edge` when Keycloak integration is merged.

File: markdown/documentation/score/installation/configuration/authentication.md [hunk 2]
```diff
@@ -1,69 +1,129 @@
...
+
+# Logging
+LOGGING_LEVEL_ORG_SPRINGFRAMEWORK_WEB=INFO
+LOGGING_LEVEL_BIO_OVERTURE_SCORE_SERVER=INFO
+LOGGING_LEVEL_ROOT=INFO
+
+# ============================
+# Server Authentication integration (Required)
+# ============================
+AUTH_SERVER_PROVIDER=keycloak
```
Comment (leoraba) [inline markdown/documentation/score/installation/configuration/authentication.md:75]: 👍

**Review by leoraba — DISMISSED** [review]
👍

**Review by leoraba — COMMENTED** [review]
Just added some comments. Everything else looks good 👍

File: markdown/documentation/score/docker-install/configuration/authentication.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+1. Navigate to the **Scopes tab** and select **"Create Authorization Scope"**
+2. **Create two authorization scopes**, one for **READ** and one for **WRITE** access.
+
+#### Resources
+
+Resources are objects or entities that users can interact with, such as a database, a file, or an API endpoint. When defining resources, you assign them to specific scopes, indicating what actions can be performed on those resources. For more details, checkout the following [Keycloak documentation](https://www.keycloak.org/docs/latest/authorization_services/index.html#resource)
+
+
+1. From the **Resource tab**, select **"Create Resource"**.
+2. Your first resource is generalized and will not be associated with any specific study or program. Name the resource `score`, and from the **authorization scopes field** dropdown select **"READ" and "WRITE"**.
```
[the remark this answers is not in the document]
Comment (MitchellShiell) [reply markdown/documentation/score/docker-install/configuration/authentication.md:124]: This is the general resource no?

File: markdown/documentation/score/docker-install/configuration/authentication.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+| `AUTH_SERVER_CLIENTID`                           | Required    | Client ID for Score [registered in Ego](/documentation/ego/user-guide/admin-ui/applications/). |
+| `AUTH_SERVER_CLIENTSECRET`                       | Required    | Client secret for Score [registered in Ego](/documentation/ego/user-guide/admin-ui/applications/). |
+| `AUTH_SERVER_SCOPE_DOWNLOAD_SYSTEM`              | Required    | System-level download scope using an API key. |
+| `AUTH_SERVER_SCOPE_DOWNLOAD_STUDY_PREFIX`        | Required    | Prefix for study-level download scopes. |
+| `AUTH_SERVER_SCOPE_DOWNLOAD_STUDY_SUFFIX`        | Required    | Suffix for study-level download scopes. |
+| `AUTH_SERVER_SCOPE_UPLOAD_SYSTEM`                | Required    | System-level upload scope using an API key. |
+| `AUTH_SERVER_SCOPE_UPLOAD_STUDY_PREFIX`          | Required    | Prefix for study-level upload scopes. |
+| `AUTH_SERVER_SCOPE_UPLOAD_STUDY_SUFFIX`          | Required    | Suffix for study-level upload scopes. |
+
+## JWT Profile (Optional)
```
[the remark this answers is not in the document]
Comment (MitchellShiell) [reply markdown/documentation/score/docker-install/configuration/authentication.md:259]: happily

**Review by leoraba — APPROVED** [review]
👍

**Pull request merged**